src/dftd4/model.f90

```diff
@@ -27,21 +33,21 @@ module dftd4_model
    implicit none
    private
 
-   public :: d4_model, new_d4_model, d4_ref
+   public :: base_d4_model, d4_ref
+
 
+   !> Abstract base dispersion model to evaluate C6 coefficients
+   type, abstract :: base_d4_model
 
-   !> Base D4 dispersion model to evaluate C6 coefficients
-   type :: d4_model
+      !> Number of atoms coupled to by pairwise parameters
+      integer :: ncoup
```

Comment:
This changes seems API breaking to me, we might need to bump dftd4 to version 4 for adding this.